### PR TITLE
JAMES-2904 Cassandra session user password and ssl configuration options

### PR DESCRIPTION
**Pull request by trantienduchn**
Rebased version of the #2730

It got enough approvals. Just need to be green then merge

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[fae816c7292842b187315b040d579ea4713b1bc2] [INFO] Apache James :: Server :: Cassandra - guice injection FAILURE [  02:00 h]
```
test this please

**Comment by rouazana** [comment]
same one:
```
[fae816c7292842b187315b040d579ea4713b1bc2] [INFO] Apache James :: Server :: Cassandra - guice injection FAILURE [  02:00 h]
```
could you check? are we leaking some resources?

In the mean time, test this please

**Comment by Arsnael** [comment]
For some reason, the log is the same, in the way it seems the CI didn't fire the build again...

test this please

**Comment by chibenwa** [comment]
> For some reason, the log is the same, in the way it seems the CI didn't fire the build again...

I think starting 2 Cassandra servers (one with auth, one other without) is too violent. Don't know how to handle this...

**Comment by trantienduchn** [comment]
I'm downloading log file to investigate and looks like it takes 1000 years to finish.

**Comment by trantienduchn** [comment]
I will take time to investigate the build, please don't trigger the CI

**Comment by trantienduchn** [comment]
Should be working as normal

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ResilientClusterProvider.java [hunk 1]
```diff
@@ -59,8 +59,8 @@ private ResilientClusterProvider(ClusterConfiguration configuration) {
         LOGGER.info("Trying to connect to Cassandra service at {} (list {})", LocalDateTime.now(),
             ImmutableList.copyOf(configuration.getHosts()).toString());
 
-        Cluster cluster = ClusterFactory.create(configuration);
```
Comment (chibenwa) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ResilientClusterProvider.java]: Ok, I'm sorry if that is the issue....

Comment (trantienduchn) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ResilientClusterProvider.java]: put the creating statement outside of the Callable object makes reactor retry always retry with a old cluster which is already closed from previous retry

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**